emcc.py

```diff
@@ -921,6 +916,12 @@ def check(input_file):
       # Note the exports the user requested
       shared.Building.user_requested_exports = shared.Settings.EXPORTED_FUNCTIONS[:]
 
```

Comment:
this change just moves the code to the right place - it confused me when debugging, before